BraveWallet/Crypto/Stores/SendTokenStore.swift

```diff
@@ -233,47 +266,90 @@ public class SendTokenStore: ObservableObject {
     else { return }
 
     isMakingTx = true
-    walletService.selectedCoin { [weak self] coin in
-      guard let self = self else { return }
-      self.rpcService.network(coin) { network in
-        if token.contractAddress.isEmpty {
-          let baseData = BraveWallet.TxData(nonce: "", gasPrice: "", gasLimit: "", to: self.sendAddress, value: "0x\(weiHexString)", data: .init())
+    self.rpcService.network(.eth) { network in
+      if token.contractAddress.isEmpty {
```

Comment:
Thinking about this one, I think this would be better checked as `token.contractAddress.caseInsensitiveCompare(network.nativeToken.contractAddress) == .orderedSame`. Would be a better check than in `RpcServiceExtensions.swift` for balance too which just compares symbols which aren't unique. 🤔